node/src/components/storage.rs

```diff
@@ -772,6 +814,19 @@ impl Storage {
             .transpose()
     }
 
+    /// Retrieves a single block header by deploy hash by looking it up in the index and returning
+    /// it.
+    fn get_block_header_by_deploy_hash<Tx: Transaction>(
+        &self,
+        tx: &mut Tx,
+        deploy_hash: DeployHash,
+    ) -> Result<Option<BlockHeader>, LmdbExtError> {
+        self.deploy_hash_index
+            .get(&deploy_hash)
+            .and_then(|block_hash| self.get_single_block_header(tx, block_hash).transpose())
+            .transpose()
+    }
```

Comment:
I don't think this function is necessary.  It should be sufficient to just return a block hash...